service/prometheus/service.go

```diff
@@ -0,0 +1,81 @@
+package prometheus
+
+import (
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+const (
+	prometheusReloadPath = "/-/reload"
+)
+
+type Prometheus interface {
+	Reload() error
+}
+
+type Config struct {
+	Logger micrologger.Logger
+
+	PrometheusAddress string
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Logger: nil,
+
+		PrometheusAddress: "",
+	}
+}
+
+type Service struct {
+	logger micrologger.Logger
+
+	prometheusAddress string
+}
+
+func New(config Config) (*Service, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	if config.PrometheusAddress == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.PrometheusAddress must not be empty")
+	}
+
+	u, err := url.ParseRequestURI(config.PrometheusAddress)
+	if err != nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.PrometheusAddress is invalid: %s", err)
+	}
+	u.Path = path.Join(u.Path, prometheusReloadPath)
+
+	service := &Service{
+		logger: config.Logger,
+
+		prometheusAddress: u.String(),
+	}
+
+	return service, nil
+}
+
+func (s *Service) Reload() error {
+	s.logger.Log("debug", "reloading prometheus config: %s", s.prometheusAddress)
+	if _, err := http.Post(s.prometheusAddress, "", nil); err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+type TestService struct{}
```

Comment:
Feel like this should be somewhere else. Needs input.

Reply:
We tend to have a `prometheustest` package with a `New` function in this case.

Reply:
+1 It's good to put that in separate package, in case it has some additional dependencies. Package containing full name suffixed with `test` is nice because it plays nicely with goimport and follows conventions of httptest package.

Reply:
Moved to `prometheustest` package. Looks good.